src/main/java/org/variantsync/diffdetective/mining/MiningTask.java

```diff
@@ -48,48 +54,49 @@ public AnalysisResult call() throws Exception {
             }
 
             /*
-             * We export all difftrees that match our filter criteria (e.g., match more than one edit class).
-             * However, we count edit classes of all DiffTrees, even those that are not exported to Linegraph.
+             * We count the edit classes of all difftrees that match our filter criteria
+             * (e.g., match more than one edit class) and export them to the destination
+             * determined by the AnalysisStrategy.
              */
+            int numDiffTrees = 0;
             final CommitDiff commitDiff = commitDiffResult.diff().get();
-            final StringBuilder lineGraph = new StringBuilder();
-            miningResult.append(LineGraphExport.toLineGraphFormat(commitDiff, lineGraph, options.exportOptions()));
-            options.analysisStrategy().onCommit(commitDiff, lineGraph.toString());
-            options.exportOptions().treeFilter().resetExplanations();
+            try (var lineGraphDestination = options.analysisStrategy().onCommit(commitDiff)) {
```

Comment:
It's not intuitive to me that an `AnalysisStrategy` always returns a stream. I see that, currently, these strategies are only used to export to file or serve as a dummy. So maybe we should rename `AnalysisStrategy` to `VariationDiffExportStrategy`?

Reply:
Well, it doesn't have to return a useful `OutputStream` (see `NullStrategy` for example). Nonetheless, this rename makes sense given the current usage pattern. I will change that in a future PR (as soon as this is merged). Should I open an issue for this (to use the issue tracker as TODO list)?